### PR TITLE
[3.9] BTS-621

### PR DESCRIPTION
**Pull request by Dronplane**
Backport of https://github.com/arangodb/arangodb/pull/15623
+ BTS-621 fix
+ unstable test fix

### CODE REVIEW

File: arangod/IResearch/IResearchLink.cpp [hunk 1]
```diff
@@ -1130,11 +1130,13 @@ Result IResearchLink::init(velocypack::Slice definition,
...
               std::to_string(_id.id()) + "'"};
     }
-    if (vocbase.server().getFeature<ClusterFeature>().isEnabled()) {
-      auto& ci = vocbase.server().getFeature<ClusterFeature>().clusterInfo();
 
-      bool const clusterWideLink =
-          _collection.id() == _collection.planId() && _collection.isAStub();
+    bool const clusterWideLink =
+        _collection.id() == _collection.planId() && _collection.isAStub();
+    auto const clusterIsEnabled =
```
Comment (MBkkt) [inline arangod/IResearch/IResearchLink.cpp:1136]: NIT bool

Comment (MBkkt) [inline arangod/IResearch/IResearchLink.cpp:1136]: NIT
bool

**Review by MBkkt — APPROVED** [review]
LGTM

**Review by MBkkt — APPROVED** [review]

**Review by gnusi — APPROVED** [review]

**Review by KVS85 — APPROVED** [review]
LGTM.

**Pull request merged**